app/parse.py

```diff
@@ -15,8 +40,76 @@ class Product:
     num_of_reviews: int
 
 
+PRODUCT_FIELDS = [field.name for field in fields(Product)]
+
+
+def get_webdriver() -> webdriver.Chrome:
+    chrome_options = Options()
+    chrome_options.add_argument("--headless")
+    chrome_options.add_argument("--disable-gpu")
+    chrome_options.add_argument("--no-sandbox")
+    driver = webdriver.Chrome(options=chrome_options)
+    # driver = webdriver.Chrome()
+    return driver
+
+
+def get_single_product(soup: Tag) -> Product:
+    return Product(
+        title=soup.select_one(".title")["title"],
+        description=soup.select_one(".description").text,
+        price=float(soup.select_one(".price").text.replace("$", "")),
+        rating=int(soup.select_one("p[data-rating]")["data-rating"]),
+        num_of_reviews=int(soup.select_one(".review-count").text.split()[0])
+    )
+
+
+def get_products(driver: webdriver.Chrome, url: str) -> list[Product]:
+    driver.get(url)
+
+    try:
+        while True:
+            button = WebDriverWait(driver, 2).until(
+                ec.element_to_be_clickable(
+                    (By.CLASS_NAME, "ecomerce-items-scroll-more"))
+            )
+            button.click()
+
+            WebDriverWait(driver, 2).until(
+                ec.presence_of_all_elements_located(
+                    (By.CLASS_NAME, "card-body"))
+            )
+
+    except (
+        TimeoutException,
+        NoSuchElementException,
+        ElementClickInterceptedException,
+        ElementNotInteractableException
+    ):
+        pass
+
+    soup = BeautifulSoup(driver.page_source, "html.parser")
+
+    return [
+        get_single_product(product_soup)
+        for product_soup in soup.select(".card-body")
+    ]
+
+
+def write_products_to_csv(products: list[Product], path: str) -> None:
+    with open(path, "w", newline="", encoding="utf-8") as file:
+        writer = csv.writer(file)
+        writer.writerow(PRODUCT_FIELDS)
+        writer.writerows([astuple(product) for product in products])
+
+
 def get_all_products() -> None:
-    pass
+    driver = get_webdriver()
+
+    for url in tqdm(URLS):
```

Comment:
Iterating over the `URLS` Enum directly is correct, but ensure that each URL is accessible and valid. If any URL is incorrect or the page structure changes, it could lead to unexpected behavior.